File: Algorithms/605. Can Place Flowers/kotlin/solution.kt
class Solution {
    fun canPlaceFlowers(flowerbed: IntArray, n: Int): Boolean {
        var counter = 0
        for (i in flowerbed.indices) {
            if (flowerbed[i] == 0 && (i == 0 || flowerbed[i-1] == 0) && (i == flowerbed.lastIndex || flowerbed[i+1] == 0)) {
                flowerbed[i] = 1
                counter++
            }
            if (counter >= n) return true
        }

        return false
    }
}

File: Algorithms/605. Can Place Flowers/swift/solution.swift
class Solution {
    func canPlaceFlowers(_ flowerbed: [Int], _ n: Int) -> Bool {
        var count = 0
        var tmpFlowerbed = flowerbed
        for i in 0..<flowerbed.count {
            if tmpFlowerbed[i] == 0 && (i == 0 || tmpFlowerbed[i - 1] == 0) && (i == flowerbed.count - 1 || tmpFlowerbed[i + 1] == 0) {
                tmpFlowerbed[i] = 1
                count += 1
            }

            if count >= n {
                return true
            }
        }

        return false
    }
}
